daemon/daemon.go

```diff
@@ -920,6 +923,23 @@ func NewDaemon(ctx context.Context, config *config.Config, pluginStore *plugin.S
 		}
 	}
 
+	sysInfo := d.RawSysInfo(false)
+	if env := os.Getenv("MOBY_DISABLE_SHIM_V2"); env != "" {
+		// MOBY_DISABLE_SHIM_V2 is deprecated from its birth.
+		// MOBY_DISABLE_SHIM_V2 is just added as a workaround for potential regression issues.
+		logrus.Warn("environment variable MOBY_DISABLE_SHIM_V2 is intended for debugging purposes, and planned to be removed in a future release")
+		d.disableShimV2, err = strconv.ParseBool(env)
+		if err != nil {
+			return nil, errors.Errorf("environment variable MOBY_DISABLE_SHIM_V2 has wrong value %q: %v", env, err)
+		}
+		if d.disableShimV2 && runtime.GOOS != "linux" {
+			return nil, errors.New("environment variable MOBY_DISABLE_SHIM_V2 is only supported on Linux")
+		}
```

Comment:
@simonferquel are environment variables shared between Windows and Linux when using WSL2 ? If so, I think we need to remove this check (or perhaps scope this whole block to linux only)

Reply:
removed the error